File: services/booking/router/driver.go
package router

import (
	"database/sql"

	"github.com/gin-gonic/gin"
	"github.com/minhthao56/monorepo-taxi/services/booking/controller"
)

func NewRouterDriver(r *gin.RouterGroup, conn *sql.DB) {
	controller := controller.NewDriverController(conn)
	r.GET("/driver/nearby", controller.FindNearByDriver)
	r.POST("/driver/update-location", controller.UpdateLocation)
	r.GET("/drivers", controller.GetDrivers)
	r.POST("/driver/update-status", controller.UpdateStatus)
}

File: clients/driver/services/booking/driver.ts
import { bookingClient } from "./client";
import type { UpdateLocationRequest } from "schema/booking/UpdateLocationRequest";
import type { UpdateStatusRequest } from "schema/booking/UpdateStatusRequest";

export async function updateLocation (req: UpdateLocationRequest) {
  return await bookingClient.post<UpdateLocationRequest, any >("/driver/update-location", req);
}

export async function updateStatus (req: UpdateStatusRequest) {
  return await bookingClient.post<UpdateStatusRequest, any >("/driver/update-status", req);
}